src/coreclr/vm/amd64/cgencpu.h

```diff
@@ -141,10 +141,10 @@ void     R8ToFPSpill(void* pSpillSlot, SIZE_T  srcDoubleAsSIZE_T)
 // Parameter size
 //**********************************************************************
 
-inline unsigned StackElemSize(unsigned parmSize, bool isValueType = false /* unused */, bool isFloatHfa = false /* unused */)
+inline uint16_t StackElemSize(unsigned parmSize, bool isValueType = false /* unused */, bool isFloatHfa = false /* unused */)
```

Comment:
I would keep it as unsigned and fix the callsite that needs it in smaller type to do the cast.

Reply:
I thought about that. However, this is a really weird API to have such loose value for. The concern I have is weird casting for a size function. I don't like casting away sizes because that breeds trouble. This approach ensures we know the size it what we want it to be. Does that convince you in this case?

Reply:
Where are we casting away the size for this one?

Reply:
Keeping this `unsigned` requires a cast at

https://github.com/dotnet/runtime/blob/ff00ac7a52225d1c528bccb174aae448542d49ac/src/coreclr/vm/mlinfo.cpp#L2976-L2985

Reply:
It is kind of strange that with this change the function gets unsigned parmSize and after aligning it returns a smaller type uint16_t.

Reply:
I've reverted it and added an assert.